File: QuickSort/QuickSort2.java
import java.io.*;
class QuickSort2
{
public static void quickSort(int x[],int lowerBound,int upperBound)
{
int stack[][],top,pp,lb,ub,e,f,g,size,m;
pp=-1;
size=upperBound-lowerBound+1;
stack=new int[size][2];
m=size-1;
top=size;
top--;
stack[top][0]=lowerBound;
stack[top][1]=upperBound;
while(top!=size)
{
lb=stack[top][0];
ub=stack[top][1];
top++;
e=lb;
f=ub;
while(e<f)
{
while(x[e]<=x[lb]) e++;
while(x[f]>x[lb]) f--;
if(e<f)
{
g=x[e];
x[e]=x[f];
x[f]=g;
}
else
{
g=x[f];
x[f]=x[lb];
x[lb]=g;
pp=f;
}
}
if(pp+1<ub)
{
top--;
stack[top][0]=pp+1;
stack[top][1]=ub;
}
if(lb<pp-1)
{
top--;
stack[top][0]=lb;
stack[top][1]=pp-1;
}
}
}
public static void main(String args[])
{
BufferedReader br=new BufferedReader(new InputStreamReader(System.in));
int x[],y,req,m;
req=5;
m=req-1;
x=new int[req];
for(y=0;y<=m;y++)
{
System.out.print("Enter a number : ");
try
{
x[y]=Integer.parseInt(br.readLine());
}catch(IOException ioException)
{
System.out.println(ioException);
return;
}catch(NumberFormatException nfe)
{
System.out.println(nfe);
return;
}
}
quickSort(x,0,m);
for(y=0;y<=m;y++) System.out.println(x[y]);
}
}

File: QuickSort/QuickSort2.py
def quickSort(x,lowerBound,upperBound):
    size=upperBound-lowerBound+1
    lbStack=[]
    ubStack=[]
    for i in range(size):
        lbStack.append(0)
        ubStack.append(0)
    top=size
    top-=1
    lbStack[top]=lowerBound
    ubStack[top]=upperBound
    while top!=size:
        lb=lbStack[top]
        ub=ubStack[top]
        top+=1
        e=lb
        f=ub
        while e<f:
            while x[e]<=x[lb] and e<ub: e+=1
            while x[f]>x[lb]: f-=1
            if e<f:
                g=x[e]
                x[e]=x[f]
                x[f]=g
            else:
                g=x[f]
                x[f]=x[lb]
                x[lb]=g
                pp=f
        if pp+1<ub:
            top-=1
            lbStack[top]=pp+1
            ubStack[top]=ub
        if lb<pp-1:
            top-=1
            lbStack[top]=lb
            ubStack[top]=pp-1
x=[]
req=5
for i in range(req): x.append(int(input("Enter a number : ")))
quickSort(x,0,req-1)
for i in x: print(i)
